File: lecture08/exercise08-A/template.cpp
// Exercise 08-A: Climbing Stairs

#include <iostream>
#include <unordered_map>

using namespace std;

// Functions

size_t count_steps(size_t n, unordered_map<size_t, size_t> &cache) {
    // TODO: Determine number of distinct ways to climb n steps using only
    // increments of 1 or 2 steps at a time.
    return 0;
}

// Main Execution

int main(int argc, char *argv[]) {
    // XXX: Because we are using size_t we will not be able to represent
    // really large numbers
    
    unordered_map<size_t, size_t> cache;
    size_t n;

    while (cin >> n) {
    	cout << count_steps(n, cache) << endl;
    }

    return 0;
}

File: lecture08/exercise08-A/template.py
#!/usr/bin/env python3

# Exercise 08-A: Climbing Stairs

import sys

# Functions

def count_steps(n: int, cache: dict[int, int]={}) -> int:
    # TODO: Determine number of distinct ways to climb n steps using only
    # increments of 1 or 2 steps at a time.
    return 0

# Main Execution

def main() -> None:
    for n in map(int, sys.stdin):
        print(count_steps(n))

if __name__ == '__main__':
    main()

# vim: set sts=4 sw=4 ts=8 expandtab ft=python:
